code-challenges.test.js

```diff
@@ -15,35 +15,142 @@
+    expect(newArray(colors1)).toEqual(expect.arrayContaining(colors1));
+    expect(newArray(colors2)).toEqual(expect.arrayContaining(colors2));
+  });
+});
+
+// ● newArray › takes in an array, removes the first item from the array and shuffles the remaining content
+//
+//    ReferenceError: newArray is not defined
+
+//const colors1 = ["purple", "blue", "green", "yellow", "pink"]
 // Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
-const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
+//const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
 // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
 
 // b) Create the function that makes the test pass.
-
+// create a function
+// use shift to remove the first item in an array
+// use sort to shuffle the array after removing the first item
+// look up different ways to use sort to shuffle the array
+
+const newArray = (array) => {
+  array.shift()
+  array.sort((a, b) => 0.6 - Math.random())
+  return array
+}
+
+// PASS  ./code-challenges.test.js
+//  newArray
+//    ✓ takes in an array, removes the first item from the array and shuffles the remaining content (2 ms)
+//
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
+// Snapshots:   0 total
+// Time:        0.325 s, estimated 1 s
 
 // --------------------2) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const nums1 = [3, 56, 90, -8, 0, 23, 6]
+
+describe('minMax', () => {
+  const nums1 = [3, 56, 90, -8, 0, 23, 6]
+  const nums2 = [109, 5, 9, 67, 8, 24]
+  it('takes an array of numbers and returns an array of the minimum and maximum numbers in that order', () => {
+    expect(minMax(nums1)).toEqual([-8, 90]);
+    expect(minMax(nums2)).toEqual([5, 109]);
+  });
+});
+
+// ● minMax › takes an array of numbers and returns an array of the minimum and maximum numbers in that order
+//
+//     ReferenceError: minMax is not defined
+
+
+//const nums1 = [3, 56, 90, -8, 0, 23, 6]
 // Expected output: [-8, 90]
-const nums2 = [109, 5, 9, 67, 8, 24]
+//const nums2 = [109, 5, 9, 67, 8, 24]
 // Expected output: [5, 109]
 
 
-// b) Create the function that makes the test pass.
+//b) Create the function that makes the test pass.
+
+// create a funstion called minMax
+// make an empty array
+// use .push to push the min and max into the empty array
+// use Math.min and Math.max to get the max and min
+// using both methods at the same time didnt work, to fix i seperated them and did two pushes first min and then max
+
+
+const minMax = (array) => {
+  let newArray = []
+  newArray.push(Math.min(...array))
+  newArray.push(Math.max(...array))
+  return newArray
+}
+
```

Comment:
Good job! I bet you can make this pseudo code even more language agnostic though...